kyo-core/shared/src/main/scala/kyo/loops.scala

```diff
@@ -220,4 +220,30 @@ object Loops:
                     ()
         loop()
     end foreach
+
+    inline def repeat[S](n: Int)(inline run: => Unit < S): Unit < S =
+        def _loop(i: Int): Unit < S =
+            loop(i)
+        @tailrec def loop(i: Int = 0): Unit < S =
+            if i == n then ()
+            else
+                run match
+                    case kyo: Kyo[Unit, S] @unchecked =>
+                        kyo.andThen(_loop(i + 1))
+                    case _ =>
+                        loop(i + 1)
+        loop()
+    end repeat
+
+    inline def forever[S](inline run: Unit < S): Unit < S =
+        def _loop(): Unit < S =
+            loop()
+        @tailrec def loop(): Unit < S =
+            run match
+                case kyo: Kyo[Unit, S] @unchecked =>
```

Comment:
Can this be `Kyo[?, S]` or `Kyo[Any, S]`? Does that impact cost of typecheck?

Reply:
We'd need to do a type cast or use `map(_ => ...` in the next line. The type itself has no cost, generic type parameters aren't checked at runtime.